source/non_matching/coupling.cc

```diff
@@ -40,6 +40,72 @@
 DEAL_II_NAMESPACE_OPEN
 namespace NonMatching
 {
+  namespace internal
+  {
+    template <int dim0, int dim1, int spacedim>
+    std::pair<std::vector<Point<spacedim>>, std::vector<unsigned int>>
+    compute_used_cells(const GridTools::Cache<dim0, spacedim> &cache,
+                       const DoFHandler<dim1, spacedim> &      immersed_dh,
+                       const Quadrature<dim1> &                quad,
+                       const Mapping<dim1, spacedim> &         immersed_mapping,
+                       const bool                              tria_is_parallel)
+    {
+      const auto &                 immersed_fe = immersed_dh.get_fe();
+      std::vector<Point<spacedim>> used_cell_ids;
+      // Keep track of which cells we actually used
+      std::vector<unsigned int> used_cells;
+      {
+        FEValues<dim1, spacedim> fe_v(immersed_mapping,
+                                      immersed_fe,
+                                      quad,
+                                      update_quadrature_points);
+        unsigned int             cell_id = 0;
+        for (const auto &cell : immersed_dh.active_cell_iterators())
+          {
+            bool use_cell = false;
+            if (tria_is_parallel)
+              {
+                const auto &bbox = cell->bounding_box();
```

Comment:
this shouldn't be a reference as the function call on the right returns an rvalue